### PR TITLE
feat: Lit@2 upgrade

### PR DESCRIPTION
**Pull request by goremikins**

### CODE REVIEW

File: packages/elements/src/tree-select/index.ts [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
 } from '@refinitiv-ui/core';
+import { customElement } from '@refinitiv-ui/core/lib/decorators/custom-element.js';
+import { property } from '@refinitiv-ui/core/lib/decorators/property.js';
+import { query } from '@refinitiv-ui/core/lib/decorators/query.js';
+import { styleMap } from '@refinitiv-ui/core/lib/directives/style-map.js';
+import { repeat } from '@refinitiv-ui/core/lib/directives/repeat.js';
+import { ifDefined } from '@refinitiv-ui/core/lib/directives/if-defined.js';
+import { VERSION } from '../version.js';
+import { CollectionComposer } from '@refinitiv-ui/utils/lib/collection.js';
+import { TimeoutTaskRunner } from '@refinitiv-ui/utils/lib/async.js';
```
[the remark this answers is not in the document]
Comment (goremikins) [reply packages/elements/src/tree-select/index.ts:15]: When importing `@refinitiv-ui/core` we do not care about tree shaking, as `@refinitiv-ui/core` includes JS code that is used essentially by all elements. When we import specifically (with js extension) we assume that only the provided file needs to be imported. This is similar to what Lit does

[the remark this answers is not in the document]
Comment (Sakchai-Refinitiv) [reply packages/elements/src/tree-select/index.ts:15]: That makes sense, Thank

File: packages/elements/src/button/index.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
+  TemplateResult
 } from '@refinitiv-ui/core';
-import { VERSION } from '..';
-
-import '../icon';
-import { registerOverflowTooltip } from '../tooltip';
+import { customElement } from '@refinitiv-ui/core/lib/decorators/custom-element.js';
+import { query } from '@refinitiv-ui/core/lib/decorators/query.js';
+import { property } from '@refinitiv-ui/core/lib/decorators/property.js';
+import { VERSION } from '../version.js';
```
[the remark this answers is not in the document]
Comment (goremikins) [reply packages/elements/src/button/index.ts:10]: re: https://lit.dev/docs/components/decorators/

```
To reduce the amount of code needed to run the component, decorators can be imported individually into component code. 
```

As I have mentioned before, public CDNs do not have tree shaking and import module as it is. So essentially if `@refinitiv-ui/core/lib/decorators/index.js ` is used, all decorators are imported in the output.

**Comment by Sakchai-Refinitiv** [comment]
Hi @goremikins,  I found this error `"elf is not defined"` when run test command on the autosuggest element. Currently, @Theeraphat-Sorasetsakul helps me test each element.
```bash
npm run test autosuggest
```
image

**Comment by goremikins** [comment]
@Sakchai-Refinitiv , autosuggest should be fixed

**Pull request merged**